### PR TITLE
[snmpwalk] adding integration

### PR DESCRIPTION
**Pull request by truthbk**
### What's this PR do?

Provides a wrapper to the snmpwalk binary.

This integration, originally written by @ross here https://gist.github.com/ross/0f6b7cc57bc3f42e93292fbafc6e1194 is a wrapper on snmpwalk and should behave similarly to the original SNMP integration (https://github.com/DataDog/integrations-core/pull/110).

### Motivation

This was written by @ross as a faster alternative to the original integration. It's merely a parser for the seriously quicker snmpwalk binary. So if you're collecting large volumes of metrics by walking, this provides a means to keep your CPU low with a much quicker check. It's not as feature rich, but may help in numerous use-cases.

This integration relies on the `snmpwalk` binary. This is a hard requirement.

### CODE REVIEW

File: snmpwalk/check.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        hostname = instance.get('metric_host', None)
+
+        # Build up our dataset
+        data = defaultdict(dict)
+        types = {}
+        for metric in metrics:
+            mib = metric['MIB']
+            table = metric['table']
+            cmd = [self.binary, '-c{}'.format(community_string),
+                   '-v2c', '-t', str(timeout), '-r', str(retries)]
```
Comment (jescholl) [inline snmpwalk/check.py]: It looks like you're hardcoding this to use snmp v2, but your example config claims to support v3.  Do you have plans to add v3 support?

**Comment by ross** [comment]
I have quite a few updates to the original version of this since it was passed along as a gist a while back. TBH it's a lot of custom stuff that probably doesn't fit/belong in a general solution, but I haven't had the time to look though and figure that out. Will this be merged or will it just live in this branch?

File: snmpwalk/check.py (hunk 1 above)
Comment (truthbk) [inline snmpwalk/check.py]: @jescholl I can definitely see that happen. Enabling the actual `snmpwalk` to make the v3 request should be straight-forward. Wondering how much work the parser might require.

**Comment by truthbk** [comment]
@ross this will be merged eventually. If you feel there's anything general you would like to contribute, it might make sense to open a PR against the active branch `jaime/snmpwrap`, and we can merge that into this one.

Thanks for your awesome work!

**Comment by truthbk** [comment]
@ross this has now been merged into the repository, so feel free to open a new PR if you wish to add new stuff! Thanks for your work! 🙇

**Pull request merged**